### ISSUE TITLE
Yocto warns about not being able to find defconfig when building `odroid-xu4`

### ISSUE THREAD

**Issue by agherzan** [resin-os/resin-odroid#60]
Running a yocto build for odroid-xu4 yields the following warning:

```
WARNING: /work/resin/resinos/resin-odroid/build/../layers/meta-odroid/recipes-kernel/linux/linux-stable_4.10.bb: Unable to get checksum for linux-stable SRC_URI entry defconfig: f
ile could not be found
```

**Comment by floion** [comment]
MACHINE=odroid-xu4 bitbake virtual/kernel
Loading cache: 100% |###############################################################################################################################################################################| Time: 0:00:00
Loaded 2767 entries from dependency cache.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION        = "1.34.0"
BUILD_SYS         = "x86_64-linux"
NATIVELSBSTRING   = "universal"
TARGET_SYS        = "arm-poky-linux-gnueabi"
MACHINE           = "odroid-xu4"
DISTRO            = "resin-os"
DISTRO_VERSION    = "2.9.7"
TUNE_FEATURES     = "arm armv7ve vfp neon callconvention-hard cortexa15"
TARGET_FPU        = "hard"
meta              
meta-poky         = "HEAD:7afc402259db8af714e3828f2cec4cc7f215ef45"
meta-oe           
meta-filesystems  
meta-networking   
meta-python       = "HEAD:dfbdd28d206a74bf264c2f7ee0f7b3e5af587796"
meta-odroid       = "HEAD:1c0210e1b3a6e96b9db17025e5f2ec35622b1755"
meta-resin-common 
meta-resin-pyro   = "HEAD:585641b7196b3ef1edddf32ff71819564aa6edc5"
meta-resin-odroid = "update_bblayers_sample:5452d9aa453ae12ab6561867b51ba8359dcbe8bf"
meta-rust         = "HEAD:94280a636c9a6dd729f9eb3ed88fe85b11ee8ffa"

Initialising tasks: 100% |##########################################################################################################################################################################| Time: 0:00:03
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
WARNING: linux-stable-4.9.24+gitAUTOINC+2f5e58ec79-r0 do_kernel_resin_checkconfig: Checking for CONFIG_DEVPTS_MULTIPLE_INSTANCES=y in the kernel configs failed for balena.
WARNING: linux-stable-4.9.24+gitAUTOINC+2f5e58ec79-r0 do_kernel_resin_checkconfig: Checking for CONFIG_DEVPTS_MULTIPLE_INSTANCES=y in the kernel configs failed for systemd.
NOTE: Tasks Summary: Attempted 1454 tasks of which 1423 didn't need to be rerun and all succeeded.

Summary: There were 2 WARNING messages shown.

I cannot reproduce on my side @agherzan 
Can you check again on a new build?

**Comment by agherzan** [comment]
Here is the log and revisions of what I tried (the current master):

```
WARNING: /work/resin/resinos/resin-odroid/build/../layers/meta-odroid/recipes-kernel/linux/linux-stable_4.10.bb: Unable to get checksum for linux-stable SRC_URI entry defconfig: file could not be found
Parsing recipes: 100% |#############################################################################################################################################| Time: 0:00:51
Parsing of 1992 .bb files complete (0 cached, 1992 parsed). 2767 targets, 360 skipped, 0 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION        = "1.34.0"
BUILD_SYS         = "x86_64-linux"
NATIVELSBSTRING   = "ubuntu-16.04"
TARGET_SYS        = "arm-poky-linux-gnueabi"
MACHINE           = "odroid-xu4"
DISTRO            = "resin-os"
DISTRO_VERSION    = "2.9.7"
TUNE_FEATURES     = "arm armv7ve vfp neon callconvention-hard cortexa15"
TARGET_FPU        = "hard"
meta              
meta-poky         = "HEAD:7afc402259db8af714e3828f2cec4cc7f215ef45"
meta-oe           
meta-filesystems  
meta-networking   
meta-python       = "HEAD:dfbdd28d206a74bf264c2f7ee0f7b3e5af587796"
meta-odroid       = "HEAD:1c0210e1b3a6e96b9db17025e5f2ec35622b1755"
meta-resin-common 
meta-resin-pyro   = "HEAD:585641b7196b3ef1edddf32ff71819564aa6edc5"
meta-resin-odroid = "master:338f268c6340417423c704481726f7fc680356e8"
```

**Comment by floion** [comment]
Why did it pick-up linux-stable_4.10.bb ? 4.10 is for Odroid C2 https://github.com/akuster/meta-odroid/blob/pyro/recipes-kernel/linux/linux-stable_4.10.bb
Your build should have used kernel version 4.9: https://github.com/akuster/meta-odroid/blob/pyro/recipes-kernel/linux/linux-stable_4.9.bb

I will retry a clean build just for checking again

**Comment by floion** [comment]
I see the warning when starting a clean build. It is basically because the BSP does not specify a compatible machine for the Odroid C2: https://github.com/akuster/meta-odroid/blob/pyro/recipes-kernel/linux/linux-stable_4.10.bb (I believe the odroid-c2 is not listed as a compatible machine there because https://github.com/akuster/meta-odroid/blob/pyro/recipes-kernel/linux/linux-hardkernel_3.14.bb is supposed to be the kernel recipe for the C2 - at least in the pyro branch of the BSP).

So because of this bitbake scans the 4.10 recipe anyway and complains there is no defconfig for the odroid-xu4, even though the 4.9 kernel recipe is to be used.

**Comment by agherzan** [comment]
We could:
1. report remote and try to get the fix in the branch we are using
2. bbappend this as a fix

**Comment by floion** [comment]
For your points:

1 - https://github.com/akuster/meta-odroid does not have issues set-up in github. I've already e-mailed the maintainer a few days ago about an actual issue on the XU4 and got no answer, so I see there's little or no interest from their part.
2 - It's a BSP issue for a machine we are not using and hence we are not impacted by this. I see not point in adding a .bbappend in our layers for something we do not use, just for fixing something in the upstream BSP. Also, the rocko branch seems to be fixing this: https://github.com/akuster/meta-odroid/tree/rocko/recipes-kernel/linux

**Comment by agherzan** [comment]
Agreed for the first point. For the second one I don't. And this is because other users are building custom images or using our build process and I'd like them to have an experience as clean as possible. And if we can fix this from our layers, I think we should.
